In [1]:
#Import libraries
import pandas as pd

In [19]:
#Load data
adoption_data = pd.read_csv('C:\\Users\\alyss\\OneDrive\\Springboard Data Science Bootcamp\\Capstone 2\\Springboard Capstone 2 Adoption Prediction Model\\pet_adoption_data.csv')
print(adoption_data.head())

FileNotFoundError: [Errno 2] No such file or directory: 'pet_adoption_data.csv'

In [6]:
#Explore the data - column types
adoption_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2007 entries, 0 to 2006
Data columns (total 13 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   PetID               2007 non-null   int64  
 1   PetType             2007 non-null   object 
 2   Breed               2007 non-null   object 
 3   AgeMonths           2007 non-null   int64  
 4   Color               2007 non-null   object 
 5   Size                2007 non-null   object 
 6   WeightKg            2007 non-null   float64
 7   Vaccinated          2007 non-null   int64  
 8   HealthCondition     2007 non-null   int64  
 9   TimeInShelterDays   2007 non-null   int64  
 10  AdoptionFee         2007 non-null   int64  
 11  PreviousOwner       2007 non-null   int64  
 12  AdoptionLikelihood  2007 non-null   int64  
dtypes: float64(1), int64(8), object(4)
memory usage: 204.0+ KB


In [20]:
# Convert generic object columns to sensible categorical data types
categorical_cols = ['PetType', 'Breed', 'Color', 'Size']
for col in categorical_cols:
    adoption_data[col] = adoption_data[col].astype('category')

# Print updated data types to verify
print(adoption_data.dtypes)

PetID                    int64
PetType               category
Breed                 category
AgeMonths                int64
Color                 category
Size                  category
WeightKg               float64
Vaccinated               int64
HealthCondition          int64
TimeInShelterDays        int64
AdoptionFee              int64
PreviousOwner            int64
AdoptionLikelihood       int64
dtype: object


In [7]:
#Explore the data - summary stats
adoption_data.describe()

,PetID,AgeMonths,WeightKg,Vaccinated,HealthCondition,TimeInShelterDays,AdoptionFee,PreviousOwner,AdoptionLikelihood
count,2007.000000,2007.000000,2007.000000,2007.000000,2007.000000,2007.000000,2007.000000,2007.000000,2007.000000
mean,1503.000000,92.279522,15.705776,0.701046,0.196313,43.974091,249.142003,0.301943,0.328351
std,579.515315,52.148363,8.327749,0.457914,0.397307,25.740253,142.887040,0.459215,0.469730
min,500.000000,1.000000,1.018198,0.000000,0.000000,1.000000,0.000000,0.000000,0.000000
25%,1001.500000,48.000000,8.730396,0.000000,0.000000,21.000000,127.000000,0.000000,0.000000
50%,1503.000000,94.000000,15.925416,1.000000,0.000000,45.000000,242.000000,0.000000,0.000000
75%,2004.500000,138.000000,22.737180,1.000000,0.000000,66.000000,375.000000,1.000000,1.000000
max,2506.000000,179.000000,29.995628,1.000000,1.000000,89.000000,499.000000,1.000000,1.000000


In [11]:
#Unique values
unique = adoption_data.nunique()
print(unique)

PetID                 2007
PetType                  4
Breed                    7
AgeMonths              179
Color                    5
Size                     3
WeightKg              2007
Vaccinated               2
HealthCondition          2
TimeInShelterDays       89
AdoptionFee            493
PreviousOwner            2
AdoptionLikelihood       2
dtype: int64


In [13]:
#Unique value frequencies
unique_freq = adoption_data.value_counts()
print(unique_freq)

PetID  PetType  Breed     AgeMonths  Color   Size    WeightKg   Vaccinated  HealthCondition  TimeInShelterDays  AdoptionFee  PreviousOwner  AdoptionLikelihood
500    Bird     Parakeet  131        Orange  Large   5.039768   1           0                27                 140          0              0                     1
1834   Rabbit   Rabbit    36         Black   Large   26.977993  1           1                86                 386          1              0                     1
1847   Cat      Siamese   35         Black   Small   12.669413  1           0                15                 165          0              0                     1
1846   Rabbit   Rabbit    164        Black   Small   14.920537  1           0                44                 181          0              0                     1
1845   Bird     Parakeet  1          Orange  Medium  21.703127  1           1                25                 279          0              1                     1
                     

In [21]:
#Calculate percent of unique values for each feature
total_rows = len(adoption_data)
percent_unique = (adoption_data.nunique() / total_rows) * 100
print("\nPercentage of Unique Values per Column:")
print(percent_unique)


Percentage of Unique Values per Column:
PetID                 100.000000
PetType                 0.199302
Breed                   0.348779
AgeMonths               8.918784
Color                   0.249128
Size                    0.149477
WeightKg              100.000000
Vaccinated              0.099651
HealthCondition         0.099651
TimeInShelterDays       4.434479
AdoptionFee            24.564026
PreviousOwner           0.099651
AdoptionLikelihood      0.099651
dtype: float64


In [9]:
#Missing values (all counts are 2007, indicating no missing values, but for the sake of completeness:)
missing = adoption_data.isna().sum()
print(missing)

PetID                 0
PetType               0
Breed                 0
AgeMonths             0
Color                 0
Size                  0
WeightKg              0
Vaccinated            0
HealthCondition       0
TimeInShelterDays     0
AdoptionFee           0
PreviousOwner         0
AdoptionLikelihood    0
dtype: int64


In [18]:
#Duplicate values
dupe = adoption_data.duplicated()
print(dupe)

print(adoption_data[adoption_data.duplicated()])

0       False
1       False
2       False
3       False
4       False
        ...  
2002    False
2003    False
2004    False
2005    False
2006    False
Length: 2007, dtype: bool
Empty DataFrame
Columns: [PetID, PetType, Breed, AgeMonths, Color, Size, WeightKg, Vaccinated, HealthCondition, TimeInShelterDays, AdoptionFee, PreviousOwner, AdoptionLikelihood]
Index: []


***Observations***  

**Age:** Ages range from 1 month to 179 months. 179 months is about 15 years, which is plausible for a senior dog. It's possible the shelter doesn't adopt out animals less than 1 month old. It's also possible that that is the floor of the data and all animals 1 month or younger are labeled as 1 month old. This is a possible limitation of the data.  

**Weight:** The minimum weight is 1.01 kg, and the maximum is 29.99 kg. For a dataset containing birds, rabbits, cats, and dogs, this wide spread is plausible. I will likely want to group or scale these later during feature engineering.

**LOS:** The max time in shelter is 89 days. This could be accurate, or it could represent a cutoff for the beginning of a new quarter, where LOS calculations are often reset. This is a possible limitation of the data.  

**Target Imbalance:** The mean of `AdoptionLikelihood` is 0.328, which tells us that roughly 33% of the entries resulted in adoptions, meaning our target classes are slightly imbalanced but highly workable for classification modeling.